pandas/core/frame.py

```diff
@@ -9838,7 +9838,27 @@ def corrwith(self, other, axis: Axis = 0, drop=False, method="pearson") -> Serie
         this = self._get_numeric_data()
 
         if isinstance(other, Series):
-            return this.apply(lambda x: other.corr(x, method=method), axis=axis)
+            if axis == 0 and method in ["pearson", "spearman"]:
```

Comment:
can you add some comments here on what you are doing and why (perf), e.g. how this is achieved

Reply:
yep! done. I also added this to the whatsnew doc under PERF in this PR